pkg/apis/monitoring/v1/types.go

```diff
@@ -1626,3 +1641,37 @@ func (l *PrometheusRuleList) DeepCopyObject() runtime.Object {
 type ProbeTLSConfig struct {
 	SafeTLSConfig `json:",inline"`
 }
+
+// Authorization contains optional `Authorization` header configuration.
+// This section is only understood by  versions of Prometheus >= 2.26.0.
+// +k8s:openapi-gen=true
+type Authorization struct {
+	// Set the authentication type. Defaults to Bearer
+	Type string `json:"type,omitempty"`
+	// Secret to read or mount
+	Credentials *v1.SecretKeySelector `json:"credentials,omitempty"`
+	// File to read a secret from
+	CredentialsFile string `json:"credentialsFile,omitempty"`
```

Comment:
One point of attention here: for security reasons, we've moved away from allowing direct access to files located on the Prometheus container's filesystem. In particular with service/pod monitors and probes, a "rogue" client with permissions to create a service monitor (for instance) could get access to the Prometheus service account's token. Instead we want the client to provision a token for us and provide it via `SecretKeySelector` (more details in the documentation of the [ArbitraryFSAccessThroughSMsConfig](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#arbitraryfsaccessthroughsmsconfig) field as well as [here](https://github.com/prometheus-operator/prometheus-operator/blob/3f047bc9c74105ba774aff726798a66b81409f7f/pkg/prometheus/operator.go#L1974-L1989)).

 My recommendation would be that for service/pod monitors and probes, we don't expose a `CredentialsFile` field. For the Prometheus resource, it makes sense to expose it though since the person managing the Prometheus resource has already access to the credentials. In practice, it would mean to create 2 structs (just like we have already with `TLSConfig` and `SafeTLSConfig`).

Reply:
Thanks, I wasn't aware of that. Two structs sounds like the way to go, I'll also add a test to `testForArbitraryFSAccess`.

Reply:
Ok I added `SafeAuthentication` and I think I changed all the types you referenced accordingly. If you could double check I'd be much obliged. I added the `alertmanger` endpoints too.